services/core-web/src/components/mine/Permit/Search/components/SearchResults.tsx

```diff
@@ -0,0 +1,187 @@
+
+interface SelectedFilter {
+    category: string;
+    value: string;
+}
+
+const SearchResults: React.FC = () => {
+    const dispatch = useAppDispatch();
+    const results = useAppSelector(selectSearchResults);
+    const loading = useAppSelector(selectSearchLoading);
+    const selectedFilters = useAppSelector(selectSearchFilters);
+
+    const [pendingFilters, setPendingFilters] = useState<SelectedFilter[]>(selectedFilters);
+    const [filtersVisible, setFiltersVisible] = useState(false);
+    const [hasFilterChanges, setHasFilterChanges] = useState(false);
+
+    useEffect(() => {
+        setPendingFilters(selectedFilters);
+    }, [selectedFilters]);
+
+    const handleFilterChange = (category: string, value: string, checked: boolean) => {
+        let newFilters: SelectedFilter[];
+        if (checked) {
+            newFilters = [...pendingFilters, { category, value }];
+        } else {
+            newFilters = pendingFilters.filter(f => !(f.category === category && f.value === value));
+        }
+        setPendingFilters(newFilters);
+        setHasFilterChanges(true);
+    };
+
+    const removeFilter = (category: string, value: string) => {
+        const updatedFilters = pendingFilters.filter(
+            f => !(f.category === category && f.value === value)
+        );
+        setPendingFilters(updatedFilters);
+        dispatch(setFilters(updatedFilters)); // Apply immediately when removing
+        setHasFilterChanges(false);
+    };
+
+    const applyFilters = () => {
+        dispatch(setFilters(pendingFilters));
+        setHasFilterChanges(false);
+    };
+
+    const clearAllFilters = () => {
+        setPendingFilters([]); // Clear pending filters (checkboxes)
+        setHasFilterChanges(true); // Show apply button
+        // Don't dispatch to Redux yet - wait for user to click Apply
+    };
+
+    if (!results && !loading) {
+        return (
+            <Row className="permit-search__empty-state" justify="center" align="middle">
+                <Empty
+                    image={Empty.PRESENTED_IMAGE_SIMPLE}
+                    description="Enter a search term to find permit conditions"
+                />
+            </Row>
+        );
+    }
+
+    const handleTagFilter = (category: string, value: string) => {
+        if (!pendingFilters.some(f => f.category === category && f.value === value)) {
+            setPendingFilters(prev => [...prev, { category, value }]);
+            setHasFilterChanges(true); // Add this line to enable the Apply button
+        }
+    };
+
+    const renderFacets = () => {
+        if (!results?.allFacets) return null;
+
+        return Object.entries(results.allFacets).map(([facetKey, facets]) => {
+            // Get current facet counts
+            const currentFacets = results.facets?.[facetKey] || [];
+
+            // Update counts while preserving all options
+            const updatedFacets = facets.map(facet => ({
```

Comment:
Not sure of the scale of things here, but it might be beneficial to memoize this.